Gem/Code/Source/Components/NetworkTeleportComponent.cpp

```diff
@@ -23,120 +24,109 @@ namespace MultiplayerSample
     {
         if (auto serializationContext = azrtti_cast<AZ::SerializeContext*>(reflection))
         {
-            serializationContext->Class<NetworkTeleportComponent, Component>()
-                ->Field("Destination", &NetworkTeleportComponent::m_reset)
-                ->Version(3);
-
-            if (AZ::EditContext* editContext = serializationContext->GetEditContext())
-            {
-                editContext->Class<NetworkTeleportComponent>("Network Teleport Component",
-                    "Teleports colliding player entities to a fixed location")
-                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
-                    ->Attribute(AZ::Edit::Attributes::Category, 
-                        "MultiplayerSample")
-                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, 
-                        AZ_CRC_CE("Game"))
-                    ->DataElement(0, &NetworkTeleportComponent::m_reset,
-                        "Destination", "Entity at which to teleport the colliding player")
-                    ;
-            }
+            serializationContext->Class<NetworkTeleportComponent, NetworkTeleportComponentBase>()
+                ->Version(4);
+            NetworkTeleportComponentBase::Reflect(reflection);
         }
     }
 
-    NetworkTeleportComponent::NetworkTeleportComponent()
-        : m_enterTrigger([this](
-            AzPhysics::SimulatedBodyHandle bodyHandle, 
+    void NetworkTeleportComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        AZ_TracePrintf("Teleporter", "Client is activating...\n");
+    }
+
+    void NetworkTeleportComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    // Controller
+
+    NetworkTeleportComponentController::NetworkTeleportComponentController(NetworkTeleportComponent& parent)
+        : NetworkTeleportComponentControllerBase(parent)
+        , m_enterTrigger([this](
+            AzPhysics::SimulatedBodyHandle bodyHandle,
             const AzPhysics::TriggerEvent& triggerEvent)
             {
                 this->OnTriggerEnter(bodyHandle, triggerEvent);
             })
     {
     }
 
-    void NetworkTeleportComponent::Activate()
+    void NetworkTeleportComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
+        AZ_TracePrintf("Teleporter", "Controller is activating...\n");
         auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
-         if (physicsSystem)
+        const AZ::EntityId self = GetEntity()->GetId();
```

Comment:
Minor: Could move self to inside if (physicsSystem)
I think in other instances you call this selfId, so could do that here for consistency too.